packages/plugins/src/foundry/utils/index.ts

```diff
@@ -27,8 +27,8 @@ export const getBuildInfo = (
 ): BuildInfo => {
   // Find the correct build info file
   for (const input of buildInfos) {
-    if (input?.output?.sources[sourceName] !== undefined) {
-      validateBuildInfo(input)
+    if (input?.output?.contracts[sourceName] !== undefined) {
+      validateBuildInfo(input, 'foundry')
```

Comment:
During partial rebuilds, Foundry may sometimes generate build info files that have lots of contracts listed as sourced but which do not include those same contracts in the output (b/c they've already been compiled so there's no need). This means we need to find the correct build info file for each contract by searching for the first file which has the target contract listed in the outputs.

Reply:
> During partial rebuilds, Foundry may sometimes generate build info files that have lots of contracts listed as sources but do not include those same contracts in the output (b/c they've already been compiled so there's no need).

Sounds good to me. Fyi, this is precisely why Foundry's bug causes an error in `getOpenZeppelinStorageLayout`. The reason that it errors is because the `buildInfo.output.sources` for these contracts looks like this:
```
      "../contracts/contracts/ChugSplashManager.sol": {
        "id": 1,
        "ast": {}
      },
```

Reply:
Interesting, since the previous build info files contain this information don't we technically have the necessary info to reconstruct the complete build info ourselves?

Reply:
We don't because we need the AST IDs, which differ across build infos